server/src/test/java/org/elasticsearch/action/search/SearchPhaseControllerTests.java

```diff
@@ -247,6 +247,7 @@ public void testMerge() {
                 0,
                 true,
                 InternalAggregationTestCase.emptyReduceContextBuilder(),
+                null,
```

Comment:
would we want to unit test the new rank codepath here?

Reply:
Added a test for this.